GitUI/CommandsDialogs/RepoHosting/ViewPullRequestsForm.cs

```diff
@@ -270,11 +297,25 @@ private void _discussionWB_DocumentCompleted(object sender, WebBrowserDocumentCo
 
         private void LoadDiffPatch()
         {
-            AsyncLoader.DoAsync(
-                () => _currentPullRequestInfo.DiffData,
-                SplitAndLoadDiff,
-                ex => MessageBox.Show(this, _strFailedToLoadDiffData.Text + Environment.NewLine + ex.Exception.Message,
-                                    _strError.Text));
+            ThreadHelper.JoinableTaskFactory.RunAsync(
+                async () =>
+                {
+                    try
+                    {
+                        await TaskScheduler.Default;
```

Comment:
:bulb: Should not need to switch to a background thread before invoking an asynchronous operation.

Reply:
Do you say this because the actual CPU time to set up the async IO should be minimal and non-blocking?